src/MessageReader.js

```diff
@@ -39,22 +39,33 @@ class StandardTypeReader {
     this.view = new DataView(buffer.buffer, buffer.byteOffset);
   }
 
+  json(): { [key: string]: any } {
+    const resultString = this.string();
+    try {
+      return JSON.parse(resultString);
+    } catch {
+      return { err: `Could not parse ${resultString}` };
```

Comment:
```suggestion
      return `Could not parse ${resultString}`;
```

no need to make it an object.